### PR TITLE
Retrieve document template content for a selected template CEMS-2071

### PR DESCRIPTION
**Pull request by steve-giles**
This update provides the ability to retrieve template content by document type and template key for a selected template.

The frontend dts service was modified with a backend call which conforms to the following specification:
https://homeceu.github.io/dts-docs/#/Templates/getTemplateBodyByKey

### CODE REVIEW

File: src/app/dts/dts.component.html [hunk 1]
```diff
@@ -11,6 +11,7 @@
   [headerHeight]="50"
   [footerHeight]="50"
   [rowHeight]="50"
+  (activate)='rowClick($event)'
```
Comment (alcfeoh) [inline src/app/dts/dts.component.html]: I'm being nitpicky, but this attribute is the only one with single quotes. All of the other ones use double quotes.

File: src/app/dts/dts.service.ts [hunk 2]
```diff
@@ -45,4 +45,15 @@ export class DtsService {
...
       );
   }
+
+  /**
+   * Gets an HTML template body by document type and template key
+   * @param documentType currently limited to 'enrollment'
+   * @param templateKey unique template identifier
+   */
+  public getTemplateByKey(documentType: string, templateKey: string): Observable<string> {
+    const url = this.url + 'template/' + documentType + '/' + templateKey;
```
Comment (alcfeoh) [inline src/app/dts/dts.service.ts]: This string might be easier to read if built with backticks and template expressions:
```
const url = `${this.url}template/${documentType}/${templateKey}` 
```

File: src/test/templates.ts [hunk 3]
```diff
@@ -1,3 +1,8 @@
+export const template = '<html>\n<head>\n<style>\n.page {\npadding-left: 10px;\npadding-right: 10px;\npage-break-after: auto;\n}\n ' +
+  '</style>\n</head>\n<body>\n<div id="container" class="page">\n<p style="text-align:center;">Enrollment #: <span>{{ enrollmentId }}' +
+  '</span></p>\n<p><span class="char-style-override-2" style="line-height: 1.2;">Course Certificate</span></p>\n</body>\n</html>';
```
Comment (alcfeoh) [inline src/test/templates.ts]: This one would definitely be easier to read with backticks and template expressions :-)
See my previous comment for an example.

**Review by alcfeoh — COMMENTED** [review]
I added a couple of comments, especially around simplifying your string building syntax.

File: src/app/dts/dts.component.html (hunk 1 above)
Comment (steve-giles) [reply src/app/dts/dts.component.html]: Fixed.

File: src/app/dts/dts.service.ts (hunk 2 above)
Comment (steve-giles) [reply src/app/dts/dts.service.ts]: Fixed.

File: src/test/templates.ts (hunk 3 above)
Comment (steve-giles) [reply src/test/templates.ts]: Great idea to make my "template" a template expression.  Using that approach I was able to mock an entire template which is readable.

**Review by alcfeoh — APPROVED** [review]

**Pull request merged**